plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1753,4 +1753,253 @@ private Set<String> getAllFilesFromCdfDirectory(String tableName)
+        assertUpdate(session, "CREATE TABLE %s (x INTEGER, part INTEGER) WITH (partitioned_by = ARRAY['part'])".formatted(tableName));
+
+        assertUpdate(session, "INSERT INTO " + tableName + "(x, part) values(1, 11)", 1);
+        assertUpdate(session, "INSERT INTO " + tableName + "(x, part) values(2, 22)", 1);
+
+        @Language("RegExp")
+        String expectedMessageRegExp = "Filter required on test_schema\\." + tableName + " for at least one partition column: part";
+        assertQueryFails(session, "ANALYZE " + tableName, expectedMessageRegExp);
+        assertQueryFails(session, "EXPLAIN ANALYZE " + tableName, expectedMessageRegExp);
+
+        assertUpdate(session, "DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testPartitionFilterMultiplePartition()
+    {
+        String tableName = "test_partition_filter_multiple_partition_" + randomNameSuffix();
+        Session session = sessionWithPartitionFilterRequirement();
+
+        assertUpdate(session, "CREATE TABLE %s (x VARCHAR, part1 INT, part2 INT) WITH (PARTITIONED_BY = ARRAY['part1', 'part2'])".formatted(tableName));
+
+        assertUpdate(session, "INSERT INTO " + tableName + "(x, part1, part2) VALUES " +
+                              """
+                                 ('a', 1, 1),
+                                 ('a', 1, 2),
+                                 ('a', 2, 1),
+                                 ('a', 2, 2),
+                                 ('b', 1, 1),
+                                 ('b', 1, 2),
+                                 ('b', 2, 1),
+                                 ('b', 2, 2)
+                                 """, 8);
+
+        assertQuery(session, "SELECT count(*) FROM %s WHERE part1 = 1".formatted(tableName), "VALUES 4");
+        assertQuery(session, "SELECT count(*) FROM %s WHERE part2 = 1".formatted(tableName), "VALUES 4");
+        assertQuery(session, "SELECT count(*) FROM %s WHERE part1 = 1 AND part2 = 2".formatted(tableName), "VALUES 2");
+        assertQuery(session, "SELECT count(*) FROM %s WHERE part2 IS NOT NULL".formatted(tableName), "VALUES 8");
+        assertQuery(session, "SELECT count(*) FROM %s WHERE part2 IS NULL".formatted(tableName), "VALUES 0");
+        assertQuery(session, "SELECT count(*) FROM %s WHERE part2 < 0".formatted(tableName), "VALUES 0");
+        assertQuery(session, "SELECT count(*) FROM %s WHERE part1 = 1 OR part2 > 1".formatted(tableName), "VALUES 6");
+        assertQuery(session, "SELECT count(*) FROM %s WHERE part1 = 1 AND part2 > 1".formatted(tableName), "VALUES 2");
+        assertQuery(session, "SELECT count(*) FROM %s WHERE part1 IS NOT NULL OR part2 > 1".formatted(tableName), "VALUES 8");
+        assertQuery(session, "SELECT count(*) FROM %s WHERE part1 IS NOT NULL AND part2 > 1".formatted(tableName), "VALUES 4");
+        assertQuery(session, "SELECT count(*) FROM %s WHERE x = 'a' AND part2 = 2".formatted(tableName), "VALUES 2");
```

Comment:
Can we apply any function on top of the partition column like concatenating multiple columns or some specific function.